content/en/tracing/troubleshooting/agent_apm_resource_usage.md

```diff
@@ -0,0 +1,44 @@
+---
+title: APM Agent Resource Usage
+kind: Documentation
+aliases:
+  - /tracing/troubleshooting/agent_apm_resource_usage
+---
+
+## How the APM Agent works
+
+The APM Agent is CPU-bound and its CPU usage is correlated with the number of spans received per second.
+
+The APM Agent is designed to buffer unprocessed payloads in memory to avoid data loss. Consequently, throttling the Agent process because of an insufficient CPU limit can lead to out of memory.
+
+## Detect out-of-CPU
+
+The following metrics emitted by APM Agent can help detect out-of-CPU:
+- `datadog.trace_agent.cpu_percent`: CPU usage (in decicores), for example, 50 (half a core), 200 (two cores). It can be compared to the [maximum CPU percentage][1] configured for the APM Agent.
+- `datadog.trace_agent.ratelimit`: If lower than 1, it means payloads are being refused due to high resource usage (cpu or memory).
```

Comment:
Perhaps this needs to be corrected a bit. Payloads are not being refused due to high resource usage. This has historically been a great source of confusion to customers and I wouldn't want them to think "why are my payloads being rate limited due to high memory usage when the APM agent is using 500 MB of memory but my machine has 2 GB". This can happen when the default value of `DD_APM_MAX_MEMORY=500` in non-Kubernetes environments.

I'd rather say something like:
```suggestion
- `datadog.trace_agent.ratelimit`: If lower than 1, payloads are being rejected due to resource usage being higher than the Datadog Agent's [configured rate limits][1] allow.
```

Reply:
P.S. (sorry for the multiple comments) I don't think that this metric goes into the "detect out of CPU" list, since it helps with figuring out if configuration limits are reached. There may still be plenty CPU on the machine.

Reply:
Good point, thanks for clarifying! In this case I think it's best to not mention `datadog.trace_agent.ratelimit` at all on this page, wdyt?

Reply:
If this page is meant to address the situation where customers keep reporting OOM to later find out it's because of out-of-CPU in fact, then I agree. Otherwise, it could be a separate section. But yes they are different things.

Reply:
Alright then, I'll remove it. Thanks for the feedback!